src/service/extensions-impl.js

```diff
@@ -454,6 +457,20 @@ export class Extensions {
 
       // Install CSS.
       const promise = this.preloadExtension(extensionId).then(extension => {
+        // An extension element/service that's declared in an FIE but _not_
+        // in the parent is not immediately installed. See addDocFactory().
+        // This is a problem for adoptServiceForEmbedIfEmbeddable(), which
+        // requires a service to be registered on an AmpDoc before adoption by
+        // the embed window.
+        // To fix this, make sure the extension is installed in the AmpDoc.
+        // Ideally, we'd be able to install a service in the FIE _without_
+        // installing it in the AmpDoc. See #19344.
+        const frameElement = /** @type {!Node} */ (dev().assert(
+            childWin.frameElement, 'frameElement not found for embed'));
+        const ampdoc = getAmpdoc(frameElement);
+        this.installExtensionInDoc_(ampdoc, extensionId);
+        return extension;
+      }).then(extension => {
         // Adopt embeddable extension services.
         extension.services.forEach(service => {
           adoptServiceForEmbedIfEmbeddable(childWin, service);
```

Comment:
@dvoytenko `adoptEmbedWindow` and `installServiceInEmbedScope` are called from here. The problem is this:
https://github.com/ampproject/amphtml/blob/7731c43f2c7d01da647e31ccf61b85e773d8e4b7/src/service.js#L655-L663

`adoptServiceForEmbedIfEmbeddable` early-outs if the service is not registered on the ampdoc. 

The short fix (this PR) is to register the service on the ampdoc before this happens. This means that the parent will always have _all_ services from its children FIEs.

The long fix is to decouple this. We should make it easier to install services on the FIE (see the follow-up in PR description). 

I'd like to merge this short-fix first to unblock amp-gwd-animation and amp-bind in FIE.

Reply:
@choumx Apologies - got caught up with some urgent holiday releases. Your explanation makes sense. It does point to the further urgency to refactor FIE to assign `ampdoc` on FIE level. Can we discuss what remains of that work?

Reply:
> It does point to the further urgency to refactor FIE to assign ampdoc on FIE level.

Completely agree. My thoughts are we can quickly mitigate by fixing the user-facing issues (APIs) in the short-term. Then, more slowly refactor/simplify under the hood in the long-term.

Specifically: in the rest of Q4, I'd like to finish the follow-up tasks in this PR and refactor the service APIs as described in #16322. That should address the largest user-facing issues for "FIE don't have an AmpDoc".